File: Templates/Java/src/main/java/ElasticSearch/MultiSearchItem.java
package ElasticSearch;

import com.fasterxml.jackson.annotation.JsonProperty;
import java.util.Map;

public class MultiSearchItem<TDocument> {
    @JsonProperty("fields")
    private Map<String, Object> fields;

    @JsonProperty("hits")
    private Hit<TDocument> hits;

    @JsonProperty("max_score")
    private Double maxScore;

    @JsonProperty("num_reduce_phases")
    private Long numReducePhases;

    @JsonProperty("pit_id")
    private String pitId;

    @JsonProperty("status")
    private Integer status;

    @JsonProperty("terminated_early")
    private Boolean terminatedEarly;

    @JsonProperty("timed_out")
    private boolean timedOut;

    @JsonProperty("took")
    private long took;

    // Getters and Setters
    // ...
}

File: Templates/TypeScript/ElasticSearch/multi-search-item.ts
import { JsonProperty } from "json-object-mapper";
import { Hit } from "./hit";

export class MultiSearchItem<TDocument> {
	@JsonProperty({ required: false })
	public fields?: { [key: string]: any };

	@JsonProperty({ name: "hits" })
	public hits: Hit<TDocument>[];

	@JsonProperty({ name: "max_score" })
	public maxScore?: number;

	@JsonProperty({ name: "num_reduce_phases" })
	public numReducePhases?: number;

	@JsonProperty({ name: "pit_id" })
	public pitId?: string;

	@JsonProperty({ name: "status" })
	public status?: number;

	@JsonProperty({ name: "terminated_early" })
	public terminatedEarly?: boolean;

	@JsonProperty({ name: "timed_out" })
	public timedOut: boolean;

	@JsonProperty({ name: "took" })
	public took: number;
}
